python/src/scipp/plotting/resampling_model.py

```diff
@@ -120,22 +123,32 @@ def _call_resample(self):
             self._home = self._resampled
             self._home_params = self._resampled_params
 
+    def update_array(self, array):
+        """
+        Update the internal array with a new array.
+        """
+        self._array = array
+
+    def reset_params(self):
+        """
+        Reset view parameters to force a plot update.
+        """
+        self._home_params = None
+        self._resampled_params = None
+
 
 class ResamplingBinnedModel(ResamplingModel):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         # TODO See #1469. This is a temporary hack to work around the
         # conversion of coords to edges in model.py.
-        new_meta = {name: var for name, var in self._array.meta.items()}
-        self._array = PlotArray(data=self._array.data,
-                                masks=self._array.masks,
-                                meta=new_meta)
-        for name, var in self._array.meta.items():
+        self._array = self._array.copy(deep=False)
```

Comment:
Something looks inconsistent here: `ResamplingModel.update_array` does not make a shallow copy, but here we do in the constructor?

Reply:
I am making a shallow copy here because i am modifying the coords below, so i need a new dict in order not to overwrite the original coords with bin centers.

Reply:
I knw, my point is: Shouldn't you do the same when `update_array()` is called?

Reply:
From Slack conversation:

Neil Vaytet: Can i ask you again why we needed to switch to bin centers here? for the binned data

Simon Heybrock: Because sc.bin distinguishes between grouping (as used for the spectrum numbers) and binning (as used for TOF).
Events to not store their spectrum number, so it does not support binning into "spectrum edges".
Basically, there is only a spectrum coord for the bins, not for the events.
and in that case only grouping is supported, not binning.

Reply:
ok then I agree, i need to also do this when `update_array()` is called.